public/app/features/datasources/components/EditDataSourceActions.tsx

```diff
@@ -62,6 +65,24 @@ export function EditDataSourceActions({ uid }: Props) {
 
   return (
     <>
+      {favoriteDataSources.enabled && dataSourceInstance && !dataSourceInstance.meta.builtIn && (
+        <IconButton
+          key={`favorite-${isFavorite ? 'favorite-mono' : 'star-default'}`}
+          name={isFavorite ? 'favorite' : 'star'}
+          iconType={isFavorite ? 'mono' : 'default'}
+          onClick={() =>
+            isFavorite
+              ? favoriteDataSources.removeFavoriteDatasource(dataSourceInstance)
+              : favoriteDataSources.addFavoriteDatasource(dataSourceInstance)
+          }
+          disabled={favoriteDataSources.isLoading}
+          tooltip={
+            isFavorite
+              ? t('datasources.edit-data-source-actions.remove-favorite', 'Remove from favorites')
+              : t('datasources.edit-data-source-actions.add-favorite', 'Add to favorites')
+          }
+        />
+      )}
```

Comment:
Maybe this would be better as its own component? Would be easier to encapsulate the logic more?